iree/compiler/Conversion/Common/VectorTransferOptimization.cpp

```diff
@@ -22,7 +24,45 @@ namespace {
 
 struct VectorTransferOptimizationPass
     : public PassWrapper<VectorTransferOptimizationPass, FunctionPass> {
-  void runOnFunction() override { vector::transferOpflowOpt(getFunction()); }
+  void runOnFunction() override {
+    vector::transferOpflowOpt(getFunction());
+    // Delete potential dead alloc and associated ops after store to load
+    // forwarding.
+    eraseDeadAllocAndStores(getFunction());
+  }
+
+ private:
+  // Track temporary allocations that are never read from. If this is the case
+  // it means both the allocations and associated stores can be removed.
+  void eraseDeadAllocAndStores(FuncOp funcOp) {
+    std::vector<Operation*> opToErase;
+    funcOp.walk([&](AllocOp op) {
+      if (allUsesAreStores(op, opToErase)) {
+        opToErase.push_back(op.getOperation());
+      }
+    });
+    for (Operation* op : opToErase) {
+      op->erase();
+    }
+  }
+  // Return true if all the uses of op are either Store/transfer_write.
+  // There can be SubviewOp users as long as all its users are also
+  // StoreOp/transfer_write. If return true it also fills out the uses, if it
+  // returns false uses is unchanged.
+  bool allUsesAreStores(Operation* op, std::vector<Operation*>& uses) {
```

Comment:
Same here. These dont need to be class methods AFAICS.

Reply:
Correct.